compliance_checker/protocols/opendap.py

```diff
@@ -4,8 +4,45 @@
 
 Functions to assist in determining if the URL is an OPeNDAP endpoint
 '''
+import io
 import requests
+import urllib.parse
+import urllib.request
 
+def create_DAP_variable_str(url):
+    """
+    Create a URL-encoded string of variables for a given DAP dataset.
+    Works on OPeNDAP datasets.
+
+    Parameters
+    ----------
+    url (str): endpoint to *DAP dataset
+
+    Returns
+    -------
+    str
+    """
+
+    # get dds
+    with urllib.request.urlopen(f"{url}.dds") as resp:
+        strb = io.StringIO(resp.read().decode())
+
+    strb.seek(8) # remove "Dataset "
+    x = strb.read()
+    strb.close()
+
+    # remove beginning and ending braces, split on newlines
+    lst = list(filter(lambda x: "{" not in x and "}" not in x, x.split("\n")))
```

Comment:
Try to express this as a nested loop rather than reassigning to the same variable.  If there are issues with the parsing, it makes it much easier to track down the issues because the variable reference isn't changed.